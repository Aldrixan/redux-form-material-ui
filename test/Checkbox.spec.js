// import Checkbox from "@material-ui/core/Checkbox"
import React from "react"
import ReduxFormMaterialUICheckbox from "../src/Checkbox"
import noop from "lodash.noop"
import renderer from "react-test-renderer"

describe("Checkbox", () => {
    it("renders an unchecked Checkbox", () => {
        const component = renderer.create(
            <ReduxFormMaterialUICheckbox input={{ onChange: noop, name: "myCheckbox" }} />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("renders a checked Checkbox", () => {
        const component = renderer.create(
            <ReduxFormMaterialUICheckbox input={{ value: true, onChange: noop, name: "myCheckbox" }} />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("should ignore defaultChecked", () => {
        const component = renderer.create(
            <ReduxFormMaterialUICheckbox input={{ onChange: noop, name: "myCheckbox" }} defaultChecked={true} />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
