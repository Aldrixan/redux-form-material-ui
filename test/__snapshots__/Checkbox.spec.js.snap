// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders a checked Checkbox 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiButtonBase-root-20 MuiIconButton-root-14 MuiSwitchBase-root-10 MuiCheckbox-root-5 MuiCheckbox-colorSecondary-9 MuiSwitchBase-checked-11 MuiCheckbox-checked-6"
    onBlur={[Function]}
    onCheck={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-19"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-23"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <input
        checked={true}
        className="MuiSwitchBase-input-13"
        name="myCheckbox"
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </span>
  <span
    className="MuiTypography-root-37 MuiTypography-body1-46 MuiFormControlLabel-label-4"
  />
</label>
`;

exports[`Checkbox renders an unchecked Checkbox 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiButtonBase-root-20 MuiIconButton-root-14 MuiSwitchBase-root-10 MuiCheckbox-root-5 MuiCheckbox-colorSecondary-9"
    onBlur={[Function]}
    onCheck={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-19"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-23"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        checked={false}
        className="MuiSwitchBase-input-13"
        name="myCheckbox"
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </span>
  <span
    className="MuiTypography-root-37 MuiTypography-body1-46 MuiFormControlLabel-label-4"
  />
</label>
`;

exports[`Checkbox should ignore defaultChecked 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiButtonBase-root-20 MuiIconButton-root-14 MuiSwitchBase-root-10 MuiCheckbox-root-5 MuiCheckbox-colorSecondary-9"
    onBlur={[Function]}
    onCheck={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-19"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-23"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        checked={false}
        className="MuiSwitchBase-input-13"
        name="myCheckbox"
        onChange={[Function]}
        type="checkbox"
      />
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </span>
  <span
    className="MuiTypography-root-37 MuiTypography-body1-46 MuiFormControlLabel-label-4"
  />
</label>
`;
